storm/String.cpp

```diff
@@ -10,6 +10,7 @@
 
 #if defined(WHOA_SYSTEM_WIN)
 #include <string.h>
+#include <Windows.h>
```

Comment:
Hmm, is this being used in `String.cpp` right now?

Reply:
Yes, but only because of the Windows error codes defined in WinError.h. Should I make a new file and define error codes in it?

Reply:
I think we could eliminate this by restoring `#include <winerror.h>` in the headers for `Error.hpp`. Sorry for being nitpicky– I'd like to try keep `#include <windows.h>` confined to implementations when possible since it drags in so many things (albeit less with `WIN32_LEAN_AND_MEAN`).